fusion/src/cli_reference.md

```diff
@@ -0,0 +1,497 @@
+
+## Commands
+
+### eval
+
+Evaluates Fusion expressions provided as command-line arguments.
+
+**Syntax:** `eval EXPRESSIONS`
+
+**Description:**
+Evaluates the given expressions as top-level Fusion forms. If the result of the last expression is not void, it's written to standard output using Ion format.
+
+**Examples:**
+```bash
+# Simple arithmetic
+fusion eval '(+ 1 2 3)'
+# Output: 6
+
+# Data construction
+fusion eval '{ name: "Alice", age: 30, active: true }'
+# Output: {name:"Alice",age:30,active:true}
+
+# Multiple expressions
+fusion eval '(define x 10) (define y 20) (+ x y)'
+# Output: 30
+
+# Working with lists
+fusion eval '[1, 2, 3, (* 2 2), (+ 3 2)]'
+# Output: [1,2,3,4,5]
+```
+
+**Shell Quoting:**
+Be careful with shell quoting and escaping:
+```bash
+# Good - single quotes protect from shell interpretation
+fusion eval '(define message "Hello World") message'
+
+# Good - escaped double quotes
+fusion eval "(define message \"Hello World\") message"
+
+# Bad - shell will interpret variables and quotes
+fusion eval (define message "Hello $USER") message
+```
+
+**Error Handling:**
+- Empty expressions will show usage information
+- Syntax errors display the problematic location
+- Runtime errors show stack traces with context
+
+### load
+
+Loads and evaluates Fusion scripts from files.
+
+**Syntax:** `load FILE`
+
+**Description:**
+Reads and evaluates a Fusion script from the specified file. The file must be readable and contain valid Fusion code. If the last expression returns a non-void value, it's written to standard output.
+
+**Examples:**
+```bash
+# Load a simple script
+fusion load hello.fusion
+
+# Load with input redirection
+echo "42" | fusion load process-number.fusion
+
+# Load multiple scripts in sequence
+fusion load setup.fusion ';' load main.fusion
+```
+
+**File Requirements:**
+- File must exist and be readable
+- File should contain valid Fusion syntax
+- File path can be relative or absolute
+
+**Sample Script (`hello.fusion`):**
+```fusion
+// Simple greeting script
+(define (greet name)
+  (string_append "Hello, " name "!"))
+
+(greet "World")
+```
+
+**Error Handling:**
+- File not found or not readable will show clear error message
+- Syntax errors in the file will show line numbers and context
+- Runtime errors preserve stack trace information
+
+### repl
+
+Starts an interactive Read-Eval-Print Loop for exploring Fusion interactively.
+
+**Syntax:** `repl`
+
+**Description:**
+Enters an interactive console where you can type Fusion expressions and see results immediately. The REPL maintains state between expressions, allowing you to define variables and functions that persist throughout the session.
+
+**Features:**
+- Colored output (blue for prompts, red for errors)
+- Command history (when using a proper terminal)
+- Built-in help system
+- Graceful error handling
```

Comment:
Remove these kinds of bullets; this isn't a marketing document.

We should also not say things like "graceful error handling" which is a matter of opinion.